pallets/nfts/src/features/approvals.rs

```diff
@@ -122,12 +122,19 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			}
 		}
 
+		// Cannot revoke approval for a specific collection item if the delegate has
+		// permission to transfer all collection items owned by the origin.
+		ensure!(
+			!CollectionApprovals::<T, I>::contains_key((collection, &details.owner, &delegate)),
+			Error::<T, I>::DelegateApprovalConflict
+		);
+
 		details.approvals.remove(&delegate);
 		Item::<T, I>::insert(collection, item, &details);
 
 		Self::deposit_event(Event::ApprovalCancelled {
 			collection,
-			item,
+			item: Some(item),
```

Comment:
Not tested.

Reply:
Resolved by adding check for `cancel_approval_works`